packages/common/infra/src/op/__tests__/client.spec.ts

```diff
@@ -0,0 +1,219 @@
+import { afterEach } from 'node:test';
+
+import { beforeEach, describe, expect, it, vi } from 'vitest';
+
+import { OpClient } from '../client';
+import type { MessageHandlers } from '../message';
+import { Op } from '../types';
+
+declare module 'vitest' {
+  interface TestContext {
+    producer: OpClient;
+    handlers: MessageHandlers;
+    postMessage: ReturnType<typeof vi.fn>;
+  }
+}
+
+class AddOp extends Op<{ a: number; b: number }, number> {
+  protected override getId(): string {
+    return 'add';
```

Comment:
The generated id should be uniq per Op instance. It is hard to know if two ids will collide. Imagine a client is called with id="add" twice, the first call will never be resolved.

Reply:
the op id is auto generated and unique, I will add checker in both client and consumer